tests/out/wgsl/operators.wgsl

```diff
@@ -32,175 +32,164 @@ fn bool_cast(x: vec3<f32>) -> vec3<f32> {
+    let unnamed_84 = (2 | 1);
+    let unnamed_85 = (2u | 1u);
+    let unnamed_86 = (vec2<i32>(2) | vec2<i32>(1));
+    let unnamed_87 = (vec3<u32>(2u) | vec3<u32>(1u));
+    let unnamed_88 = (2 & 1);
+    let unnamed_89 = (2u & 1u);
+    let unnamed_90 = (vec2<i32>(2) & vec2<i32>(1));
+    let unnamed_91 = (vec3<u32>(2u) & vec3<u32>(1u));
+    let unnamed_92 = (2 ^ 1);
+    let unnamed_93 = (2u ^ 1u);
+    let unnamed_94 = (vec2<i32>(2) ^ vec2<i32>(1));
+    let unnamed_95 = (vec3<u32>(2u) ^ vec3<u32>(1u));
+    let unnamed_96 = (2 << 1u);
+    let unnamed_97 = (2u << 1u);
+    let unnamed_98 = (vec2<i32>(2) << vec2<u32>(1u));
+    let unnamed_99 = (vec3<u32>(2u) << vec3<u32>(1u));
+    let unnamed_100 = (2 >> 1u);
+    let unnamed_101 = (2u >> 1u);
+    let unnamed_102 = (vec2<i32>(2) >> vec2<u32>(1u));
+    let unnamed_103 = (vec3<u32>(2u) >> vec3<u32>(1u));
 }
 
 fn comparison() {
-    let unnamed_112 = (2 == 1);
-    let unnamed_113 = (2u == 1u);
-    let unnamed_114 = (2.0 == 1.0);
-    let unnamed_115 = (vec2<i32>(2) == vec2<i32>(1));
-    let unnamed_116 = (vec3<u32>(2u) == vec3<u32>(1u));
-    let unnamed_117 = (vec4<f32>(2.0) == vec4<f32>(1.0));
-    let unnamed_118 = (2 != 1);
-    let unnamed_119 = (2u != 1u);
-    let unnamed_120 = (2.0 != 1.0);
-    let unnamed_121 = (vec2<i32>(2) != vec2<i32>(1));
-    let unnamed_122 = (vec3<u32>(2u) != vec3<u32>(1u));
-    let unnamed_123 = (vec4<f32>(2.0) != vec4<f32>(1.0));
-    let unnamed_124 = (2 < 1);
-    let unnamed_125 = (2u < 1u);
-    let unnamed_126 = (2.0 < 1.0);
-    let unnamed_127 = (vec2<i32>(2) < vec2<i32>(1));
-    let unnamed_128 = (vec3<u32>(2u) < vec3<u32>(1u));
-    let unnamed_129 = (vec4<f32>(2.0) < vec4<f32>(1.0));
-    let unnamed_130 = (2 <= 1);
-    let unnamed_131 = (2u <= 1u);
-    let unnamed_132 = (2.0 <= 1.0);
-    let unnamed_133 = (vec2<i32>(2) <= vec2<i32>(1));
-    let unnamed_134 = (vec3<u32>(2u) <= vec3<u32>(1u));
-    let unnamed_135 = (vec4<f32>(2.0) <= vec4<f32>(1.0));
-    let unnamed_136 = (2 > 1);
-    let unnamed_137 = (2u > 1u);
-    let unnamed_138 = (2.0 > 1.0);
-    let unnamed_139 = (vec2<i32>(2) > vec2<i32>(1));
-    let unnamed_140 = (vec3<u32>(2u) > vec3<u32>(1u));
-    let unnamed_141 = (vec4<f32>(2.0) > vec4<f32>(1.0));
-    let unnamed_142 = (2 >= 1);
-    let unnamed_143 = (2u >= 1u);
-    let unnamed_144 = (2.0 >= 1.0);
-    let unnamed_145 = (vec2<i32>(2) >= vec2<i32>(1));
-    let unnamed_146 = (vec3<u32>(2u) >= vec3<u32>(1u));
-    let unnamed_147 = (vec4<f32>(2.0) >= vec4<f32>(1.0));
+    let unnamed_104 = (2 == 1);
+    let unnamed_105 = (2u == 1u);
+    let unnamed_106 = (2.0 == 1.0);
+    let unnamed_107 = (vec2<i32>(2) == vec2<i32>(1));
+    let unnamed_108 = (vec3<u32>(2u) == vec3<u32>(1u));
+    let unnamed_109 = (vec4<f32>(2.0) == vec4<f32>(1.0));
+    let unnamed_110 = (2 != 1);
+    let unnamed_111 = (2u != 1u);
+    let unnamed_112 = (2.0 != 1.0);
+    let unnamed_113 = (vec2<i32>(2) != vec2<i32>(1));
```

Comment:
I think we know how many times a particular expression is used. So we could just generate `_ = xxx` if it's not used, and the name isn't specified

Reply:
I'm now thinking a new kind of statement for phony assignment might be better (i.e. `Phony(Handle<Expression>)`) since using named_expressions with _ feels like a hack. What do you think?

Reply:
It's not supposed to be a named expression at all. It's just an expression that is emitted at a particular point.
I don't think this warrants an IR change, we shouldn't be introducing a new statement type.